bayespy/inference/vmp/vmp.py

```diff
@@ -144,6 +144,11 @@ def update(self, *nodes, repeat=1, plot=False, tol=None, verbose=True):
         else:
             plot_nodes = [self[x] for x in plot]
 
+        # Make certain that at least one of the nodes in the update list
+        # has been observed
+        if ~np.any([n.observed for n in nodes]):
```

Comment:
Also, I realized that some observations must be in the model but not necessarily in the set of nodes that are currently updated.